dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java

```diff
@@ -804,7 +806,76 @@ private String getPotentialKeyFromCategory(Category category) {
 
 	}
 
-	/**
+	@CloseDBIfOpened
+	@Override
+	public Category findByVariable(final String variable, final User user,
+								   final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
+
+		final Category category = categoryFactory.findByVar(variable);
+
+		if(!InodeUtils.isSet(category.getCategoryId()))
+			return null;
+
+		if(!permissionAPI.doesUserHavePermission(category, PermissionAPI.PERMISSION_USE, user, respectFrontendRoles))
+			throw new DotSecurityException("User doesn't have permission to use this category = " +
+					category.getInode());
+
+		return category;
+	}
+
+    @Override
+    public List<Category> getCategoriesFromContent(Contentlet contentlet, User user, boolean respectFrontendRoles)

```

Comment:
this is a new method that looks like could return null, we prefer optional rather than null